composer/trainer/trainer_hparams.py

```diff
@@ -343,13 +343,22 @@ def initialize_object(self) -> Trainer:
 
         try:
             steps_per_epoch = len(train_dataloader)
+            if self.train_subset_num_batches is not None:
+                steps_per_epoch = self.train_subset_num_batches
         except (AttributeError, NotImplementedError):
             steps_per_epoch = None
 
         batch_size = None
         if train_dataloader.batch_size is not None:
             batch_size = train_dataloader.batch_size * dist.get_world_size()
 
+        # Adjust tokens per epoch based on train_subset_num_batches
+        if self.train_subset_num_batches is not None and batch_size is not None and steps_per_epoch is not None:
+            if tokens_per_epoch is not None:
+                tokens_per_epoch = batch_size * self.train_subset_num_batches
```

Comment:
I don't think this is correct, since each batch will have a different number of tokens. But nothing is using this right now...

Reply:
@moinnadeem, is there an easy solution to this?